src/nodes.coffee

```diff
@@ -404,6 +404,13 @@ exports.InfinityLiteral = class InfinityLiteral extends NumberLiteral
   compileNode: ->
     [@makeCode '2e308']
 
+exports.NaNLiteral = class NaNLiteral extends NumberLiteral
+  constructor: ->
+    super 'NaN'
+
+  compileNode: ->
+    [@makeCode '(0/0)']
```

Comment:
I went with the simple approach of _always_ wrapping in parentheses here. Do we want to only do it if strictly needed? (If so, how?)